### PR TITLE
88 delete/donors/id

### PR DESCRIPTION
**Pull request by DeenaAlghazali**

### CODE REVIEW

**Review by HaniOlwan — APPROVED** [review]

File: server/controllers/admin/donors/deleteDonorById.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+const deleteDonorById = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { id } = await paramsSchema.validateAsync(req.params);
+    const donorInfo = await Donor.findByPk(id);
+    if (!donorInfo) throw new CustomError('ID does not exist', 400);
+    await Donor.destroy({
+      where: { id },
+    });
+    res.status(200).json({ message: 'Deleted successfully' });
```
Comment (Raghad-Mezied) [inline server/controllers/admin/donors/deleteDonorById.ts:14]: As 200 is the default so you can don't send it.

File: server/routes/adminDonor.ts [hunk 2]
```diff
@@ -0,0 +1,8 @@
+import { Router } from 'express';
+import { deleteDonorById } from '../controllers';
+
+const adminDonorRouter = Router();
+
+adminDonorRouter.route('/donor/:id').delete(deleteDonorById);
```
Comment (Raghad-Mezied) [inline server/routes/adminDonor.ts:6]: Again is it allowed for anyone to delete a doner?

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: server/controllers/admin/donors/deleteDonorById.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+import { Donor } from '../../../database/models';
+import { CustomError, paramsSchema } from '../../../utils';
+
+const deleteDonorById = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { id } = await paramsSchema.validateAsync(req.params);
+    const deletedDonor = await Donor.destroy({
+      where: { id },
+    });
+    if (!deletedDonor) throw new CustomError('ID does not exist', 400);
```
Comment (Raghad-Mezied) [inline server/controllers/admin/donors/deleteDonorById.ts:11]: This message will be shown to the user, so add a more user-friendly message.

File: server/tests/route.test.ts [hunk 4]
```diff
@@ -427,6 +427,40 @@ describe('GET/campaigns', () => {
...
+  test('case: succeeded | Deleted successfully', async () => {
+    const donorId = 2;
+    const response = await request(app)
+      .delete(`/api/admin/donor/${donorId}`)
+      .set('Cookie', [
+        'ACCESS_TOKEN=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwibmFtZSI6ImFkbWluIiwiaXNBZG1pbiI6dHJ1ZSwiaWF0IjoxNjUxOTk4NDgzLCJleHAiOjE2NTQ1OTA0ODN9.LBvMMkPbcTeBMbKBeOQ7sYe1s-Wy5zHjhbjjTtcByFw',
+      ])
+      .expect(200);
+    expect(response.body.message).toBe('Deleted successfully');
+  });
```
Comment (Raghad-Mezied) [inline server/tests/route.test.ts:440]: Test the case with an unauthorized user

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by HaniOlwan — APPROVED** [review]

File: server/controllers/admin/donors/deleteDonorById.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+import { Donor } from '../../../database/models';
+import { CustomError, paramsSchema } from '../../../utils';
+
+const deleteDonorById = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const { id } = await paramsSchema.validateAsync(req.params);
+    const deletedDonor = await Donor.destroy({
+      where: { id },
+    });
+    if (!deletedDonor) throw new CustomError('The donor you are trying to delete does not exist', 400);
```
Comment (Mohammed-Hamada) [inline server/controllers/admin/donors/deleteDonorById.ts:11]: `doesn't`

File: server/controllers/admin/donors/deleteDonorById.ts [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    res.json({ message: 'Deleted successfully' });
```
Comment (Mohammed-Hamada) [inline server/controllers/admin/donors/deleteDonorById.ts:12]: `Donor deleted successfully`

File: server/controllers/admin/donors/deleteDonorById.ts [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  } catch (error) {
+    if (error.name === 'ValidationError') next(new CustomError(error.details[0].message, 400));
```
Comment (Mohammed-Hamada) [inline server/controllers/admin/donors/deleteDonorById.ts:14]: just you want to send `error.message`, that equals `error.details[0].message`

**Review by Mohammed-Hamada — CHANGES_REQUESTED** [review]

**Review by farah2003 — APPROVED** [review]

**Review by Mohammed-Hamada — APPROVED** [review]

**Pull request merged**